generatedocs.py

```diff
@@ -0,0 +1,424 @@
+from git import Repo # git for downloading assets
+from jinja2 import Environment, FileSystemLoader # template magic
+import json # reading the input file
+import os # file paths
+import re # regex for searching for assets files
+from tqdm import tqdm # progress bar
+
+################################################################################
+#                      ASSET COMPONENTS HELPER FUNCTIONS                       #
+################################################################################
+
+def clone_assets_folder_git(folder_name):
+  """
+  Clones asset directory from OpenSpace git repository
+  Uses the latest master
+  Returns absolute path to the asset folder
+  """
+  def print_progress(op_code, cur_count, max_count=None, message=""):
+    print(message)
+
+  print("Cloning assets examples...")
+  repo = Repo.init(folder_name)
+  branch_name = "origin/master"
+
+  # Create a new remote if there isn't one already created
+  origin = repo.remotes[0] if len(repo.remotes) > 0 else None
+  if not origin:
+    print("No remote origin found. Creating OpenSpace remote...")
+    origin = repo.create_remote("origin", "https://github.com/OpenSpace/OpenSpace")
+
+  print(f"Fetching OpenSpace on branch {branch_name}... this might take a while")
+  origin.fetch(progress=print_progress)
+
+  data_assets_path = "data/assets" 
+  git = repo.git()
+  git.checkout(branch_name, "--", data_assets_path)
```

Comment:
```suggestion
  print("Cloning assets examples...")
  repo = Repo.init(folder_name)

  # Create a new remote if there isn't one already created
  origin = repo.remotes[0] if len(repo.remotes) > 0 else None
  if not origin:
    print("No remote origin found. Creating OpenSpace remote...")
    origin = repo.create_remote("origin", "https://github.com/OpenSpace/OpenSpace")

  print(f"Fetching OpenSpace... this might take a while")
  origin.fetch(progress=print_progress, depth=1)

  data_assets_path = "data/assets"
  git = repo.git()
  git.checkout("--", data_assets_path)
```

The `fetch` will default to the `master` branch, so no need to specify that separately.  The big change is the addition of `depth=1` which will make a shallow copy that speeds up the fetch dramatically